src/Microsoft.Health.Dicom.Core/Features/Update/UpdateInstanceService.cs

```diff
@@ -0,0 +1,202 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.IO;
+using DicomFileExtensions = Microsoft.Health.Dicom.Core.Features.Retrieve.DicomFileExtensions;
+
+namespace Microsoft.Health.Dicom.Core.Features.Update;
+
+public class UpdateInstanceService : IUpdateInstanceService
+{
+    private readonly IFileStore _fileStore;
+    private readonly IMetadataStore _metadataStore;
+    private readonly ILogger<UpdateInstanceService> _logger;
+    private readonly RecyclableMemoryStreamManager _recyclableMemoryStreamManager;
+    private const int LargeObjectsizeInBytes = 1024 * 1024;
+    private const int StageBlockSizeInBytes = 4 * 1024 * 1024;
```

Comment:
comment 4MB? Do these need to be configurable?

Reply:
Yes, it can be.